File: reactdapercore.client/src/classes/ToDoItem.ts
// eslint-disable-next-line @typescript-eslint/no-unused-vars
export default class TodoItem
{   
    id: string;
    name: string;
    description: string;
    // Duration: number;
    // StartDate: Date;
    // SetReminder: boolean;
    constructor(id: string, name: string, description: string
        
        // Duration: number, 
        // StartDate: Date, 
        // SetReminder: boolean
    )
    {
        this.id = id;
        this.name = name;
        this.description = description;
        // this.Duration = Duration;
        // this.StartDate = StartDate;
        // this.SetReminder = SetReminder;
    }
}

File: Models/DbModels/ToDoItem.cs
using System.ComponentModel.DataAnnotations;

namespace Models.DbModels
{
    public class ToDoItem
    {
        public int Id { get; set; }

        [StringLength(255)]
        public string Description { get; set; }

        public int Duration { get; set; }

        public bool SetReminder { get; set; }

        public DateTime Date { get; set; }
    }
}
